src/Symfony/Component/Translation/IdentityTranslator.php

```diff
@@ -25,17 +25,19 @@ class IdentityTranslator implements LegacyTranslatorInterface, TranslatorInterfa
     use TranslatorTrait;
 
     private $selector;
+    private $debug;
 
     /**
-     * @param MessageSelector|null $selector The message selector for pluralization
+     * @param bool $debug
      */
-    public function __construct(MessageSelector $selector = null)
+    public function __construct(MessageSelector $messageSelector = null, $debug = true)
     {
-        $this->selector = $selector;
-
-        if (__CLASS__ !== \get_class($this)) {
-            @trigger_error(sprintf('Calling "%s()" is deprecated since Symfony 4.2.', __METHOD__), E_USER_DEPRECATED);
+        if ($debug instanceof MessageSelector) {
+            @trigger_error(sprintf('Calling "%s()" is with a "%s" or "%s" deprecated since Symfony 4.3.', __METHOD__, MessageSelector::class, null), E_USER_DEPRECATED);
```

Comment:
"is" should be moved in front of "deprecated"

Reply:
done